cluster/mesos/docker/util.sh

```diff
@@ -0,0 +1,301 @@
+function cluster::mesos::docker::run_in_docker {
+  entrypoint="$1"
+  if [[ "${entrypoint}" = "./"* ]]; then
+    # relative to project root
+    entrypoint="/go/src/github.com/GoogleCloudPlatform/kubernetes/${entrypoint}"
+  fi
+  shift
+  args="$@"
+
+  container_id=$(
+    docker run \
+      -d \
+      -e "KUBERNETES_PROVIDER=${KUBERNETES_PROVIDER}" \
+      -v "${KUBE_ROOT}:/go/src/github.com/GoogleCloudPlatform/kubernetes" \
+      -v "$(dirname ${KUBECONFIG}):/root/.kube" \
+      -v "/var/run/docker.sock:/var/run/docker.sock" \
+      --link docker_mesosmaster1_1:mesosmaster1 \
+      --link docker_mesosslave1_1:mesosslave1 \
+      --link docker_mesosslave2_1:mesosslave2 \
+      --link docker_apiserver_1:apiserver \
+      --entrypoint="${entrypoint}" \
+      mesosphere/kubernetes-mesos-test \
+      ${args}
+  )
+
+  docker logs -f ${container_id} &
+
+  # trap and kill for better signal handing
+  trap 'echo "Killing container ${container_id}" 1>&2 && docker kill ${container_id}' INT TERM
+  exit_status=$(docker wait ${container_id})
+  trap - INT TERM
+
+  if [ "$exit_status" != 0 ]; then
+    echo "Exited ${exit_status}" 1>&2
+  fi
+
+  docker rm -f ${container_id} > /dev/null
+
+  return ${exit_status}
+}
+
+# Generate kubeconfig data for the created cluster.
+function create-kubeconfig {
+  local kubectl="${KUBE_ROOT}/cluster/kubectl.sh"
+
+  export CONTEXT="${KUBERNETES_PROVIDER}"
+  export KUBECONFIG=${KUBECONFIG:-$DEFAULT_KUBECONFIG}
+  # KUBECONFIG determines the file we write to, but it may not exist yet
+  if [[ ! -e "${KUBECONFIG}" ]]; then
+    mkdir -p $(dirname "${KUBECONFIG}")
+    touch "${KUBECONFIG}"
+  fi
+
+  local token="$(cut -d, -f1 ${provider_root}/certs/apiserver/tokens)"
+  "${kubectl}" config set-cluster "${CONTEXT}" --server="${KUBE_SERVER}" --certificate-authority="${provider_root}/certs/root-ca.crt"
+  "${kubectl}" config set-context "${CONTEXT}" --cluster="${CONTEXT}" --user="cluster-admin"
+  "${kubectl}" config set-credentials cluster-admin --token="${token}"
+  "${kubectl}" config use-context "${CONTEXT}" --cluster="${CONTEXT}"
+
+   echo "Wrote config for ${CONTEXT} to ${KUBECONFIG}" 1>&2
+}
+
+# Perform preparations required to run e2e tests
+function prepare-e2e {
+  echo "TODO: prepare-e2e" 1>&2
+}
+
+# Execute prior to running tests to build a release if required for env
+function test-build-release {
+  # Make a release
+  export KUBERNETES_CONTRIB=mesos
+  export KUBE_RELEASE_RUN_TESTS=N
+  "${KUBE_ROOT}/build/release.sh"
+}
+
+# Must ensure that the following ENV vars are set
+function detect-master {
+  #  echo "KUBE_MASTER: $KUBE_MASTER" 1>&2
+
+  docker_id=$(docker ps --filter="name=docker_apiserver" --quiet)
+  if [[ "${docker_id}" == *'\n'* ]]; then
+    echo "ERROR: Multiple API Servers running in docker" 1>&2
+    return 1
+  fi
+
+  details_json=$(docker inspect ${docker_id})
+  master_ip=$(jq '.[0].NetworkSettings.IPAddress' --raw-output <<< "$details_json")
```

Comment:
do we require python anywhere else? otherwise isn't that a dependency of its own?